apps/health_campaign_field_worker_app/lib/pages/language_selection.dart

```diff
@@ -36,56 +45,80 @@ class LanguageSelectionPage extends StatelessWidget {
                   return const Offstage();
                 }
 
-                return BlocBuilder<LocalizationBloc, LocalizationState>(
+                return BlocConsumer<LocalizationBloc, LocalizationState>(
+                  listener: (context, state) {
+                    if (state.loading && !isDialogVisible) {
+                      isDialogVisible = true;
+                      DigitComponentsUtils().showLocationCapturingDialog(
+                        context,
+                        '',
+                        DigitSyncDialogType.inProgress,
+                      );
+                    } else if (!state.loading && isDialogVisible) {
+                      isDialogVisible = false;
+                      DigitComponentsUtils().hideLocationDialog(context);
+                    }
+                  },
                   builder: (context, localizationState) {
+                    if (localizationState.loading && !isDialogVisible) {
+                      WidgetsBinding.instance.addPostFrameCallback((_) {
+                        isDialogVisible = true;
+                        DigitComponentsUtils().showLocationCapturingDialog(
```

Comment:
Replace this with a common dialog instead of using locationCapture

Reply:
created a util function to show a localization loader